apps/src/lab2/levelEditors/aichatSettings/ModelSelectionFields.tsx

```diff
@@ -0,0 +1,105 @@
+import FieldSection from './FieldSection';
+import {Visibility} from '@cdo/apps/aichat/types';
+
+const modelDropdownItems = modelDescriptions.map(model => {
+  return {value: model.id, text: model.name};
+});
+
+const ModelSelectionFields: React.FunctionComponent = () => {
+  const {setModelSelectionValues, aichatSettings} = useContext(UpdateContext);
+  const shouldDisableAdditionalModelSelection =
+    aichatSettings.visibilities.selectedModelId !== Visibility.EDITABLE;
+  const [additionalAvailableModelIds, setAdditionalAvailableModelIds] =
+    useState<Set<string>>(
+      new Set(
+        aichatSettings.availableModelIds.filter(
+          id => id !== aichatSettings.initialCustomizations.selectedModelId
+        )
+      )
+    );
+  const selectedModelId = aichatSettings.initialCustomizations.selectedModelId;
+
+  const onDropdownChange = useCallback(
+    (e: React.ChangeEvent<HTMLSelectElement>) => {
+      setModelSelectionValues(additionalAvailableModelIds, e.target.value);
+    },
+    [additionalAvailableModelIds, setModelSelectionValues]
+  );
+
+  const onCheckboxChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      if (e.target.checked) {
+        setAdditionalAvailableModelIds(
+          additionalAvailableModelIds.add(e.target.name)
+        );
+        setModelSelectionValues(additionalAvailableModelIds, selectedModelId);
+      } else {
+        additionalAvailableModelIds.delete(e.target.name);
+        setAdditionalAvailableModelIds(additionalAvailableModelIds);
+        setModelSelectionValues(additionalAvailableModelIds, selectedModelId);
+      }
+    },
+    [additionalAvailableModelIds, selectedModelId, setModelSelectionValues]
+  );
+
+  return (
+    <FieldSection
+      fieldName="selectedModelId"
+      labelText="Model Selection"
+      description="The initial model selected for the chatbot."
+      inputType="custom"
+      inputNode={
+        <>
+          <SimpleDropdown
+            labelText=""
+            onChange={onDropdownChange}
+            items={modelDropdownItems}
+            selectedValue={selectedModelId}
+            name="model"
+            size="s"
+          />
+          <br />
+          <BodyFourText>
+            <i>
+              Models available to the student to select from and compare to each
+              other. If making this setting editable, it's best practice to
+              select at least 2 models so students have something to compare -
+              otherwise, consider making this read only or hidden.
+            </i>
+          </BodyFourText>
+          {modelDescriptions.map(model => {
+            return (
+              <div key={model.id} className={moduleStyles.fieldRow}>
+                <Checkbox
+                  key={model.id}
+                  name={model.id}
+                  label={model.name}
+                  checked={
+                    additionalAvailableModelIds.has(model.id) ||
```

Comment:
I fixed this in a different way so we don't save unnecessary stuff in the aiCustomizations in that case. See line 27-33

Reply:
Yeah I think I'm going to punt this one and log a follow-up bug to be triaged. I think I'd need a bigger refactor to move the `additionalAvailableModelIds` up to the parent component, because when I try to reset the array based on `shouldDisableAdditionalModelSelection` I get React errors because it has to re-render while it's already re-rendering.  I don't want to just hide them in the way that you suggested, because it's going to save that list of values even if it's not showing it, and I'd rather show an honest view of what will be saved.

Reply:
SGTM! Thanks for thinking this through.